### PR TITLE
Improve ScanDetail and ExecDetails

### PR DESCRIPTION
**Pull request by breeswish**
1. Improvement: Supply many comments for ScanDetailV2 and TimeDetail fields.

2. Improvement: Rename message `HandleTime` to `TimeDetail`, field `handle_time` to `time_detail`, `bool handle_time` to `bool record_time_stat`, etc.

   Field and message rename is upgrade compatible. However code need to be updated accordingly.

3. Add: In GetResponse and BatchGetResponse, use `ExecDetailsV2` instead of `ScanDetailV2`, so that time can be reported as well in future, as proposed in https://github.com/tikv/tikv/issues/8942.

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -817,39 +821,76 @@ enum IsolationLevel {
...
-    int64 wait_ms = 1;
-    // Processing time, excluding wait time.
-    int64 process_ms = 2;
+message TimeDetail {
+    // Off-cpu wall time elapsed in TiKV side. Usually this includes queue waiting time and
+    // other kind of waitings in series.
+    int64 wait_wall_time_ms = 1;
+    // Off-cpu and on-cpu wall time elapsed to actually process the request payload. It does not
+    // include `wait_wall_time`.
+    // This field is very close to the CPU time in most cases. Some wait time spend in RocksDB
```
Comment (zhongzc) [inline proto/kvrpcpb.proto:830]: Do we really have the ability to measure the CPU time of a single request, even we don't take Mutex wait time into account?

Comment (breeswish) [reply proto/kvrpcpb.proto:830]: The precise CPU time can be measured by user_cpu_time + system_cpu_time. Not sure about the cost though.

Comment (zhongzc) [reply proto/kvrpcpb.proto:830]: IMO, this field is not close to CPU time. It's a typical wall time.

Comment (breeswish) [reply proto/kvrpcpb.proto:830]: Oh yes. I only want to express that, in 99% cases, the higher this field is, the higher CPU usage there will be.

**Review by zhongzc — APPROVED** [review]

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -817,39 +821,76 @@ enum IsolationLevel {
...
 message ExecDetails {
-    HandleTime handle_time = 1;// set when ctx.handle_time = true or meet slow query
-    ScanDetail scan_detail = 2;// set when ctx.scan_detail = true or meet slow query
-    bool use_scan_detail_v2 = 3;// set when ctx.scan_detail = true or meet slow query
-    ScanDetailV2 scan_detail_v2 = 4;// set when ctx.scan_detail = true or meet slow query
+    // Available when ctx.record_time_stat = true or meet slow query.
+    TimeDetail time_detail = 1;
+
+    // Available when ctx.record_scan_stat = true or meet slow query.
+    ScanDetail scan_detail = 2;
```
Comment (zhongzc) [inline proto/kvrpcpb.proto:885]: Maybe `reserved`s are needed here.

Comment (breeswish) [reply proto/kvrpcpb.proto:885]: Good catch

**Comment by breeswish** [comment]
@crazycs520 PTAL if you are free, thanks!

**Review by crazycs520 — APPROVED** [review]
LGTM

File: proto/kvrpcpb.proto [hunk 3]
```diff
@@ -36,8 +36,9 @@ message GetResponse {
     bytes value = 3;
     // True if the key does not exist in the database.
     bool not_found = 4;
-    // Set if ctx.scan_detail = true or meet slow query.
-    ScanDetailV2 scan_detail_v2 = 5;
+    reserved 5;
```
Comment (sticnarf) [inline proto/kvrpcpb.proto:39]: This field is not used in a released version. Do we really need to reserve it?

Comment (breeswish) [reply proto/kvrpcpb.proto:39]: Only for our lovely developers who are using the master branch.

**Pull request merged**